TODO.org

```diff
@@ -34,18 +34,68 @@
 
 * Completion
 
-See https://github.com/renpy/vscode-language-renpy/blob/master/src/completion.ts for
-inspiration:
+- [X] local label completion
 
-- [ ] local/global label completion
+- [X] local image completion
 
-- [ ] keyword-specific completions
+- [X] local hide completion
+
+- [-] local transform completion
+
+  - [X] single transform
+
+  - [ ] dot-separated identifiers
+
+  - [ ] multiple comma-separated transforms
+
+  - [ ] built-in transforms
+
+- [ ] image directory completion
+
+- [ ] keyword completion
+
+- [ ] global completion
+
+** Further notes
+
+On every project compilation Ren'Py builds a file =game/saves/navigation.json=. The file
+contains project meta data as well as lists of symbol definition files/lines grouped by
+type: callables, transforms, labels, screens, defines. VSCode uses the file to support
+definition search and completion.
+
+It should be easy to watch the file using the file notification facility
+(https://www.gnu.org/software/emacs/manual/html_node/elisp/File-Notifications.html) and
+maintain a project-wide symbol index on the Emacs side. This simplifies both project-wide
+symbol completion and definition lookups.
```

Comment:
If you want to try incorporating data that Ren'Py can dump from the game, I think the command to do that would be:
```sh
renpy --json-dump-common --json-dump-private --json-dump /path/to/output-file.json /path/to/project-directory quit
```
...maybe it is too slow to be useful. If it is possible to integrate it, how much of a help or hiderence it may be could vary per file, so perhaps there would need to be a way to configure the completions sources. As a third option, I imagine it would also be possible to generate a TAGS file much faster than waiting for the Ren'Py compilation or data dump.

Reply:
I'll test both. 

My gut feeling is that fast TAGS regex-based generation is worse than running renpy. Two benefts of relying on Renpy: 

- renpy compile is precise
- the user would have to comple the game anyway.

VSCode assumes the existance of navigation.json anyway.